dns/gcloud/dns.tf

```diff
@@ -0,0 +1,89 @@
+provider "google" {
+  version = "< 3.0.0"
+}
+
+variable "name" {
+}
+
+variable "project" {
+}
+
+variable "zone_name" {
+}
+
+variable "domain" {
+}
+
+data "google_dns_managed_zone" "domain" {
+    name    = var.zone_name
+    project = var.project
+}
+
+variable "public_ip" {
+  type = list(string)
+}
+
+variable "rsa_public_key" {
+}
+
+data "external" "key2fp" {
+  program = ["python", "${path.module}/key2fp.py"]
+  query = {
+    ssh_key = var.rsa_public_key
+  }
+}
+
+resource "google_dns_record_set" "loginX_A" {
+  count        = length(var.public_ip)
+  managed_zone = data.google_dns_managed_zone.domain.name
+  project      = var.project
+  name         = join("", [var.name, format("%d", count.index + 1), ".", var.name, ".", var.domain, "."])
```

Comment:
Since it is for login nodes, I was thinking we could rename the record to

```
login, format("%d", count.index + 1), ".", var.name, ".", var.domain, "."
```

This would break with what is currently done in Compute Canada, but it would make more sense with the subdomain rule.